features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/prerequisite/MemoryRequirementCheckerTest.kt

```diff
@@ -0,0 +1,121 @@
+
+    private lateinit var testedMemoryRequirementChecker: MemoryRequirementChecker
+
+    @TempDir
+    lateinit var tempDir: File
+
+    private lateinit var fakeMemoryFile: File
+
+    @Mock
+    private lateinit var mockInternalLogger: InternalLogger
+
+    @IntForgery(min = 0)
+    private var fakeMemorySizeInMb: Int = 0
+
+    @BeforeEach
+    fun `set up`() {
+        fakeMemoryFile = File(tempDir, "meminfo")
+        testedMemoryRequirementChecker = MemoryRequirementChecker(
+            minRamSize = fakeMemorySizeInMb,
+            memInfoFile = fakeMemoryFile,
+            internalLogger = mockInternalLogger
+        )
+    }
+
+    @Test
+    fun `M have correct name W call name()`() {
+        assertThat(testedMemoryRequirementChecker.name()).isEqualTo("ram")
+    }
+
+    @Test
+    fun `M return true W check memory size meet configuration`() {
+        // Given
+        val fakeMemoryFileContent = generateFakeMemoryFileContent(fakeMemorySizeInMb + 1)
+        testedMemoryRequirementChecker = MemoryRequirementChecker(
+            minRamSize = fakeMemorySizeInMb,
+            memInfoFile = fakeMemoryFile,
+            internalLogger = mockInternalLogger
+        )
+        // When
+        fakeMemoryFile.writeText(fakeMemoryFileContent)
+        val result = testedMemoryRequirementChecker.checkMinimumRequirement()
+
+        // Then
+        assertThat(result).isTrue()
+    }
+
+    @Test
+    fun `M return false W check cpu core number not meet configuration`() {
+        // Given
+        val fakeMemoryFileContent = generateFakeMemoryFileContent(fakeMemorySizeInMb - 1)
+        testedMemoryRequirementChecker = MemoryRequirementChecker(
+            minRamSize = fakeMemorySizeInMb,
+            memInfoFile = fakeMemoryFile,
+            internalLogger = mockInternalLogger
+        )
+
+        // When
+        fakeMemoryFile.writeText(fakeMemoryFileContent)
+        val result = testedMemoryRequirementChecker.checkMinimumRequirement()
+
+        // Then
+        assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `M return correct checked value W check check`() {
+        // Given
+        val fakeMemoryFileContent = generateFakeMemoryFileContent(fakeMemorySizeInMb)
+        testedMemoryRequirementChecker = MemoryRequirementChecker(
+            minRamSize = fakeMemorySizeInMb,
+            memInfoFile = fakeMemoryFile,
+            internalLogger = mockInternalLogger
+        )
+
+        // When
+        fakeMemoryFile.writeText(fakeMemoryFileContent)
+        testedMemoryRequirementChecker.checkMinimumRequirement()
+
+        // Then
+        assertThat(testedMemoryRequirementChecker.checkedValue()).isEqualTo((fakeMemorySizeInMb).toLong())
+    }
+
+    private fun generateFakeMemoryFileContent(memorySizeInMb: Int): String {
+        return mapOf<String, Any>(
+            "MemTotal" to (memorySizeInMb.toLong() * 1000)
+        ).map { (key, value) -> "$key:\t$value" }
+            .joinToString("\n")
```

Comment:
I'm a bit confused here: it seems that map has only one key-value pair, so the whole method can be written as `return "MemTotal:\t${memorySizeInMb.toLong() * 1000}"` if I'm not wrong.

Or `return "MemTotal:\t${memorySizeInMb.toLong() * 1000}\n"` if newline at the EOF is necessary.

Reply:
Indeed, I intended to add some irrelevant lines to simulate the real file, updated.

Reply:
you can still do :)

Reply:
some irrelevant memory infos are added here